pkg/build/registry/build/strategy.go

```diff
@@ -0,0 +1,121 @@
+package build
+
+import (
+	"fmt"
+	"time"
+
+	"code.google.com/p/go-uuid/uuid"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/fielderrors"
+
+	"github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/build/api/validation"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+)
+
+// strategy implements behavior for nodes
+type strategy struct {
+	runtime.ObjectTyper
+}
+
+// Decorator is used to compute duration of a build since its not stored in etcd yet
+var Decorator = func(obj runtime.Object) error {
+	build := obj.(*api.Build)
```

Comment:
This casting without checking for correctness is the only thing that worries me the most, especially that it's repeated a couple of times in this PR. The previous version has had such a check and I'm pretty sure we should keep it here as well, just in case.